Sources/GRPCHTTP2Core/Client/GRPCClientStreamHandler.swift

```diff
@@ -67,16 +67,27 @@ extension GRPCClientStreamHandler {
       switch frameData.data {
       case .byteBuffer(let buffer):
         do {
-          try self.stateMachine.receive(buffer: buffer, endStream: endStream)
-          loop: while true {
-            switch self.stateMachine.nextInboundMessage() {
-            case .receiveMessage(let message):
-              context.fireChannelRead(self.wrapInboundOut(.message(message)))
-            case .awaitMoreMessages:
-              break loop
-            case .noMoreMessages:
-              context.fireUserInboundEventTriggered(ChannelEvent.inputClosed)
-              break loop
+          switch try self.stateMachine.receive(buffer: buffer, endStream: endStream) {
+          case .endRPCAndForwardErrorStatus(let status):
+            if let rpcError = RPCError(status: status) {
+              context.fireErrorCaught(rpcError)
+            }
```

Comment:
This isn't quite right: we should be forwarding this as a `Status` and (empty) `Metadata`.

Reply:
Ah - should we also forward a close then?

Reply:
yup